### ISSUE TITLE
eslint-plugin-awscdk: classes extending a generic base class are not recognized as Constructs

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#602]
### Describe the bug

In eslint-plugin-awscdk, a class whose inheritance chain passes through a generic class (`class Api extends BaseConstruct<ApiProps>`) is not recognized as a Construct, and neither is the instance created by `new Generic<T>(...)` / `new Api(...)`. Every type-aware rule silently skips such classes and `new` expressions. oxlint-plugin-awscdk reports them all.

A generic base class (`abstract class BaseStack<P extends BaseProps> extends Stack`, `class BaseConstruct<P> extends Construct`) is a common way to share props handling across stacks/constructs, so every subclass of such a base is unchecked in the eslint plugin.

`src/generic.ts`:

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

export interface BaseProps {
  readonly name: string;
}

export abstract class BaseConstruct<P extends BaseProps> extends Construct {
  constructor(scope: Construct, id: string, props: P) {
    super(scope, id);
    console.log(props.name);
  }
}

export interface ApiProps extends BaseProps {
  readonly unused: string;
}

export class Api extends BaseConstruct<ApiProps> {
  public bucket: Bucket;

  constructor(myScope: Construct, id: string, props: ApiProps) {
    super(myScope, id, { name: props.name });
    this.bucket = new Bucket(this, "api-bucket");
  }
}

export class Generic<T> extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
  }
}

export class Root extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    new Api(this, "api", { name: "n", unused: "u" });
    new Generic<string>(this, "generic");
  }
}
```

Rules enabled (`"error"`): `construct-constructor-property`, `no-construct-in-public-property-of-construct`, `no-mutable-public-property-of-construct`, `no-unused-props`, `pascal-case-construct-id`.

```
$ npx eslint src/generic.ts
  24:36  error  Construct ID must be PascalCase  awscdk/pascal-case-construct-id

$ npx oxlint --format=unix -c .oxlintrc.json src/generic.ts
src/generic.ts:22:15: Constructor of a class extending Construct must have the property names 'scope, id' or 'scope, id, props' [Error/awscdk(construct-constructor-property)]
src/generic.ts:20:3: Public property 'bucket' of Construct should not use Construct type 'Bucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-public-property-of-construct)]
src/generic.ts:20:3: Public property 'bucket' should be readonly. Consider adding the 'readonly' modifier. [Error/awscdk(no-mutable-public-property-of-construct)]
src/generic.ts:22:47: Property 'unused' is defined in props but never used [Error/awscdk(no-unused-props)]
src/generic.ts:24:36: Construct ID must be PascalCase. [Error/awscdk(pascal-case-construct-id)]
src/generic.ts:37:19: Construct ID must be PascalCase. [Error/awscdk(pascal-case-construct-id)]
src/generic.ts:38:31: Construct ID must be PascalCase. [Error/awscdk(pascal-case-construct-id)]
```

Line 24 is reported by eslint only because `Bucket` itself is not generic.

### Expected Behavior

eslint-plugin-awscdk reports the same seven diagnostics as oxlint-plugin-awscdk: `Api` is a Construct (its chain is `Api -> BaseConstruct<ApiProps> -> Construct`), and `new Api(...)` / `new Generic<string>(...)` create Constructs.

### Current Behavior

eslint-plugin-awscdk reports only line 24. `Api` and the two `new` expressions on lines 37-38 are treated as non-Constructs by every rule.

### Plugin

eslint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.2 (compared against oxlint-plugin-awscdk 1.0.2)

### Language Version

TypeScript 6.0.3

### Additional Context

```yaml
# machine-readable summary
rules: ["construct-constructor-property", "no-construct-in-public-property-of-construct", "no-mutable-public-property-of-construct", "no-unused-props", "pascal-case-construct-id", "every rule that goes through isExtendsFromTargetSuperClass"]
symptom: ["false-negative", "cross-plugin-divergence"]
affected_packages: ["eslint-plugin-awscdk"]
root_cause: "packages/eslint/src/core/ts-type/checker/is-extends-target-super-class.ts:26 (Type.getBaseTypes() returns undefined for a generic instantiation / TypeReference, so the base-chain walk stops there)"
verified_on: "main @ 9caea55 (eslint-plugin-awscdk 4.4.2, oxlint-plugin-awscdk 1.0.2)"
related_issues: ["#536", "#428"]
```

Root cause confirmed with the TypeScript compiler API on the fixture above (`type.getBaseTypes()` vs `checker.getBaseTypes(type.target)`):

```
class Api:          isClassOrInterface=true  getBaseTypes=BaseConstruct<ApiProps>
  -> BaseConstruct<ApiProps>: objectFlags Reference, Class:false  isClassOrInterface=false  getBaseTypes=undefined  checker.getBaseTypes(target)=Construct
new Generic<string>: objectFlags Reference, Class:false  isClassOrInterface=false  getBaseTypes=undefined  target=Generic<T>  checker.getBaseTypes(target)=Construct
new Api:            isClassOrInterface=true  getBaseTypes=BaseConstruct<ApiProps>  (stops one level up, as above)
```

`Type.getBaseTypes()` is implemented as `this.isClassOrInterface() ? checker.getBaseTypes(this) : undefined`, and a `TypeReference` created by instantiating a generic class does not carry `ObjectFlags.Class`. Verified fix direction: resolving `type.target` (when present) before calling `getBaseTypes()` in `is-extends-target-super-class.ts` makes the eslint plugin report the same seven diagnostics as oxlint on the fixture, with `examples/` still at 43 errors. `is-resource-with-readonly-interface.ts:73` uses the same call (no visible effect today because CDK resource classes are not generic).

The oxlint plugin is unaffected because corsa's `checker.getBaseTypes()` resolves references itself.

Reproduction environment:

- macOS 14.1, Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.268.0 / constructs 10.8.1
- ESLint 10.9.1 + typescript-eslint 8.69.0 + TypeScript 6.0.3
- oxlint 1.81.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.13.1)

**Comment by ren-yamanashi** [comment]
Correction to the last paragraph of the Additional Context: `is-resource-with-readonly-interface.ts:73` **is** visibly affected as well, through a user-defined generic subclass of a CDK resource.

```ts
import * as s3 from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

export class TypedBucket<T> extends s3.Bucket {
  public readonly tag?: T;
}

export interface MyConstructProps {
  readonly typed: TypedBucket<string>;
}

export class MyConstruct extends Construct {
  public readonly typed: TypedBucket<string>;
  constructor(scope: Construct, id: string, props: MyConstructProps) {
    super(scope, id);
    this.typed = props.typed;
  }
}
```

```
$ npx eslint src/typed-bucket.ts
(no output)

$ npx oxlint --format=unix -c .oxlintrc.json src/typed-bucket.ts
src/typed-bucket.ts:9:3: Interface property 'typed' should not use CDK Construct type 'TypedBucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-interface)]
src/typed-bucket.ts:13:3: Public property 'typed' of Construct should not use Construct type 'TypedBucket'. Consider using an interface or type alias instead. [Error/awscdk(no-construct-in-public-property-of-construct)]
```

`TypedBucket<string>` is a `TypeReference`, so both `isResourceType` (via `is-extends-target-super-class.ts:26`) and the `implements` walk in `is-resource-with-readonly-interface.ts:73` stop at the first `getBaseTypes()` call. Same root cause, same fix (resolve `type.target` before `getBaseTypes()`), one more call site to cover. Verified on main @ 9caea55 with the environment listed above.
